Dockerfile

```diff
@@ -1,4 +1,6 @@
-FROM ubuntu:xenial
+FROM ubuntu:focal
```

Comment:
Instead, should use the official Docker container of Python, this makes it faster as well.

Reply:
used python-slim now and removed python from apt commands
I didn't use the alpine image  because I found many recommendations against it . E.g. https://pythonspeed.com/articles/alpine-docker-python/

Reply:
That is because of the wheels, however, we have the wheels available for Alpine :)
(as Home Assistant builds them). But nice for a future update.

Reply:
to be honest I'm glad if I get python working at all and also not a docker pro  :)  So I'm pretty agnostic which images to use 

Should I rather use this for the PR ? 
A quick test didn't show a huge difference to the python-slim build but it is indeed pretty small 146MB
````
FROM python:alpine

ARG BUILD_ARCH
ENV WHEELS_LINKS=https://wheels.home-assistant.io/alpine-3.13/${BUILD_ARCH}/

RUN \
    apk add --no-cache \
    curl \
    git \
    make \
    doxygen \
    openssh-client 

RUN pip3 install --no-cache-dir --no-binary :all: \
    sphinx

EXPOSE 8000
WORKDIR /data/esphomedocs

CMD ["make", "webserver"]
````

Reply:
Nah let's stick with what we have for now, maybe in the future; We could add dev containers based on the current implementation as well (which isn't possible with Alpine).